packages/flutter/lib/src/services/text_editing_delta.dart

```diff
@@ -0,0 +1,370 @@
+  /// applied.
+  final TextRange composing;
+
+  /// This method will take the given [TextEditingValue] and return a new
+  /// [TextEditingValue] with that instance of [TextEditingDelta] applied to it.
+  TextEditingValue apply(TextEditingValue value);
+}
+
+/// The delta is inserting a single/or contigous sequence of characters.
+@immutable
+class TextEditingDeltaInsertion extends TextEditingDelta {
+  /// Creates an insertion delta for a given change to the editing state.
+  ///
+  /// {@macro flutter.services.TextEditingDelta}
+  const TextEditingDeltaInsertion({
+    required String oldText,
+    required this.textInserted,
+    required this.insertionOffset,
+    required TextSelection selection,
+    required TextRange composing,
+  }) : super(
+      oldText: oldText,
+      selection: selection,
+      composing: composing,
+  );
+
+  /// The text that is being inserted into [oldText].
+  final String textInserted;
+
+  /// The offset in the [oldText] where the insertion begins.
+  final int insertionOffset;
+
+  @override
+  TextEditingValue apply(TextEditingValue value) {
+    // To stay inline with the plain text model we should follow a last write wins
+    // policy and apply the delta to the oldText. This is due to the asyncronous
+    // nature of the connection between the framework and platform text input plugins.
+    String newText = oldText;
+    newText = _replace(newText, textInserted, insertionOffset, insertionOffset);
+    return value.copyWith(text: newText, selection: selection, composing: composing);
+  }
+}
+
+/// The delta is deleting a single/or contiguous sequence of characters.
+@immutable
+class TextEditingDeltaDeletion extends TextEditingDelta {
+  /// Creates a deletion delta for a given change to the editing state.
+  ///
+  /// {@macro flutter.services.TextEditingDelta}
+  const TextEditingDeltaDeletion({
+    required String oldText,
+    required this.deletedRange,
+    required TextSelection selection,
+    required TextRange composing,
+  }) : super(
+    oldText: oldText,
+    selection: selection,
+    composing: composing,
+  );
+
+  /// The range in [oldText] that is being deleted.
+  final TextRange deletedRange;
+
+  /// The text from [oldText] that is being deleted.
+  String get textDeleted => oldText.substring(deletedRange.start, deletedRange.end);
+
+  @override
+  TextEditingValue apply(TextEditingValue value) {
+    // To stay inline with the plain text model we should follow a last write wins
+    // policy and apply the delta to the oldText. This is due to the asyncronous
+    // nature of the connection between the framework and platform text input plugins.
+    String newText = oldText;
+    newText = _replace(newText, '', deletedRange.start, deletedRange.end);
+    return value.copyWith(text: newText, selection: selection, composing: composing);
+  }
+}
+
+/// The delta is replacing a range of characters with a new sequence of text.
```

Comment:
The phrasing of these descriptions currently assume previous knowledge of what a "delta" is.

This should be worded more generally, describing what it is and does without large assumptions of previous knowledge of the system

Reply:
This applies to the other docs for the other classes as well.

Reply:
Thanks for the feedback! I updated all the docs.